unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+ * @test_procedure Steps:
+ *                      -# Set weston output with name is "custom_screen" and id is 1
+ *                      -# Calling the output_created_event()
+ *                      -# Verification point:
+ *                         +# wl_list_insert() must be called once time
+ *                         +# wl_list_init() must be called once time
+ *                         +# weston_compositor_schedule_repaint() must be called once time
+ *                         +# The result output should same with prepare data
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(ControllerTests, output_created_event_customScreen)
+{
+    struct weston_output l_createdOutput;
+    l_createdOutput.id = 1;
+    l_createdOutput.name = (char*)"custom_screen";
+
+    output_created_event(&mp_iviShell->output_created, &l_createdOutput);
+
+    ASSERT_EQ(wl_list_insert_fake.call_count, 1);
+    ASSERT_EQ(wl_list_init_fake.call_count, 1);
+    ASSERT_EQ(weston_compositor_schedule_repaint_fake.call_count, 1);
+    struct iviscreen *lp_iviScreen = (struct iviscreen*)(uintptr_t(wl_list_init_fake.arg0_history[0]) - offsetof(struct iviscreen, resource_list));
+    ASSERT_EQ(lp_iviScreen->shell, mp_iviShell);
+    ASSERT_EQ(lp_iviScreen->output, &l_createdOutput);
+    ASSERT_EQ(lp_iviScreen->id_screen, mp_iviShell->screen_id_offset + l_createdOutput.id);
+
+    free(lp_iviScreen);
+}
+
+/** ================================================================================================
+ * @test_id             output_created_event_defaultConfigScreen
+ * @brief               Test case of output_created_event() where weston output is a default name
+ *                      same screen info default
+ * @test_procedure Steps:
+ *                      -# Set weston output with name is default name and id is 1000
+ *                      -# Calling the output_created_event()
+ *                      -# Verification point:
+ *                         +# wl_list_insert() must be called once time
+ *                         +# wl_list_init() must be called once time
+ *                         +# weston_compositor_schedule_repaint() must be called once time
+ *                         +# The result output should same with prepare data
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(ControllerTests, output_created_event_defaultConfigScreen)
+{
+    struct weston_output l_createdOutput;
+    l_createdOutput.id = 1000;
+    l_createdOutput.name = mp_screenInfo->screen_name;
+
+    output_created_event(&mp_iviShell->output_created, &l_createdOutput);
+
+    ASSERT_EQ(wl_list_insert_fake.call_count, 1);
+    ASSERT_EQ(wl_list_init_fake.call_count, 1);
+    ASSERT_EQ(weston_compositor_schedule_repaint_fake.call_count, 1);
+    struct iviscreen *lp_iviScreen = (struct iviscreen*)(uintptr_t(wl_list_init_fake.arg0_history[0]) - offsetof(struct iviscreen, resource_list));
```

Comment:
good to get lp_iviScreen pointer via wl_list_insert_fake. care about what is insert to list of screen